execution/specExecutor.go

```diff
@@ -272,25 +273,48 @@ func (e *specExecutor) executeScenarios(scenarios []*gauge.Scenario) ([]result.R
 }
 
 func (e *specExecutor) executeScenario(scenario *gauge.Scenario) (*result.ScenarioResult, error) {
-	e.currentExecutionInfo.CurrentScenario = &gauge_messages.ScenarioInfo{
-		Name:     scenario.Heading.Value,
-		Tags:     getTagValue(scenario.Tags),
-		IsFailed: false,
-	}
+	var scenarioResult *result.ScenarioResult
 
-	scenarioResult := &result.ScenarioResult{
-		ProtoScenario:             gauge.NewProtoScenario(scenario),
-		ScenarioDataTableRow:      gauge.ConvertToProtoTable(&scenario.ScenarioDataTableRow),
-		ScenarioDataTableRowIndex: scenario.ScenarioDataTableRowIndex,
-		ScenarioDataTable:         gauge.ConvertToProtoTable(&scenario.DataTable.Table),
-	}
-	if err := e.addAllItemsForScenarioExecution(scenario, scenarioResult); err != nil {
-		return nil, err
+	shouldRetry := RetryOnlyTags == ""
+
+	if !shouldRetry {
+		spec := e.specification
+		tagValues := make([]string, 0)
+		if spec.Tags != nil {
```

Comment:
Tags provided at scenario level should also be considered here. So you may have to look at `scenario.Tags` here.

Reply:
Probably, I did not get your point.

As far as I understand `filter.NewScenarioFilterBasedOnTags` accepts `specTags` and returns structure with `Filter` function which can be used to filter scenario (lines 287, 289)

Reply:
Oh, my bad.